### PR TITLE
fix: input plugin statsd bug

### PR DESCRIPTION
**Pull request by rentiansheng**
### Required for all PRs:

- [ ] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #

## fix bug:
  - fix: #10113
     reason: input plugins(statsd), parse metrics line, value has '=' parse error(return empty tag,ignore tag)

     After repair: 
 ```
       line ="uri=/service/endpoint?sampleParam={paramValue}"
       parse result  key="uri", val="/service/endpoint?sampleParam\={paramValue}"
```

### format 
file: config/config.go adjust the deprecated position, some IDE can disply strikethrough, eg: goland

### CODE REVIEW

**Comment by rentiansheng** [comment]
!signed-cla

File: config/config.go [hunk 1]
```diff
@@ -165,6 +165,7 @@ type AgentConfig struct {
 	// TODO(cam): Remove UTC and parameter, they are no longer
 	// valid for the agent config. Leaving them here for now for backwards-
 	// compatibility
+	// Deprecated: 1.0.0 after, has no effect
```
Comment (sspaink) [inline config/config.go:168]: Looks like this change duplicates a comment already there, was this change intentional?

File: plugins/inputs/statsd/statsd.go [hunk 2]
```diff
@@ -799,6 +799,12 @@ func parseKeyValue(keyvalue string) (string, string) {
 		val = split[1]
 	} else if len(split) == 1 {
 		val = split[0]
+	} else if len(split) > 2 {
```
Comment (sspaink) [inline plugins/inputs/statsd/statsd.go:802]: Can you update/add a test to cover this case? It will help verify it fixes the issue. Thanks!

**Review by sspaink — CHANGES_REQUESTED** [review]
Thank you for looking into fixing this problem, I've added two comments for you to look at.

File: config/config.go (hunk 1 above)
Comment (rentiansheng) [reply config/config.go:168]: adjust the deprecated position, some IDE can disply strikethrough, eg: goland
effect：
image

File: plugins/inputs/statsd/statsd.go (hunk 2 above)
Comment (rentiansheng) [reply plugins/inputs/statsd/statsd.go:802]: add test case for input plugins(statsd) to test parseKeyValue function

**Comment by rentiansheng** [comment]
@sspaink   TestGraphiteOkWithTagsAndSeparatorUnderscore  timeout , message panic: test timed out after 10m0s

**Comment by rentiansheng** [comment]
@sspaink    TestGraphiteOkWithTagsAndSeparatorUnderscore  has error.   message:  "panic: test timed out after 10m0s"

File: config/config.go (hunk 1 above)
Comment (sspaink) [reply config/config.go:168]: Ok cool I didn't realize Golang had that feature, can you then remove the old comment that is identical?

Comment (rentiansheng) [reply config/config.go:168]: ok

**Review by sspaink — APPROVED** [review]

**Review by powersj — APPROVED** [review]
